pxr/imaging/hdx/unitTestDelegate.cpp

```diff
@@ -58,7 +58,49 @@ TF_DEFINE_PRIVATE_TOKENS(
     _tokens,
 
     (renderBufferDescriptor)
-);    
+);
+
+
+// XXX: These mostly-empty subclasses are here to prevent a link-time
+// optimization with HdRenderIndex::InsertTask<T>(), hdx/taskController.cpp and
+// HD_TRACE_FUNCTION() for address sanitized builds of USD. Without these
+// classes the linker will incorrectly "optimize" calls to the expanded
+// HD_TRACE_FUNCTION() in GetRenderIndex()->InsertTask<HdxPickTask>() for
+// taskController.cpp.
+class Hdx_UnitTestPickTask : public HdxPickTask {
```

Comment:
Another potential solution to this problem would be to change the linking order for `hdx` to have `taskController.o` be linked before `unitTestDelegate.o`. This would present two problems:
1. The `unitTestDelegate` class would need to be moved into the `PUBLIC_CLASSES` section of `pxr_library` since private classes are configured first.
2. The link time optimization would still happen but with `uniTestDelegate.o` instead of `taskController.o`. Since `unitTestDelegate` is a helper for unit tests, disabling address sanitizer instrumentation for the offending `Add*Task(HdSceneDelegate*, const SdfPath&)` would be OK.